self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryTestCase.java

```diff
@@ -0,0 +1,86 @@
+package com.selfxdsd.core.mock;
+
+import com.selfxdsd.api.Provider;
+import com.selfxdsd.api.Storage;
+import com.selfxdsd.api.User;
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class InMemoryTestCase {
```

Comment:
@criske All classes should be either final or abstract. Test classes should always be final.

Also, you need to write a JavaDoc for the class. Something simple such as:

```java
/*
 * Unit tests for class {@link InMemoryUsers}.
 * @author name
 * @version $Id$
 * @since 0.0.1
 * @todo #16:30min Continue writing unit tests for the InMemory storage infrastructure.
 *  In the end all classes and methods should be covered with unit tests so we can rely 100%
 *  on them.
 */
```

Reply:
@criske The class javadoc is also where you add the ``TODO`` for the next task. We call these TODOs **"puzzles"** and they need to have a strict format (otherwise 0pdd will complain): 

```
@todo #16:30min Continue writing unit tests for the InMemory storage infrastructure.
 In the end all classes and methods should be covered with unit tests so we can rely 100%
 on them.
```